commcare_connect/reports/helpers.py

```diff
@@ -48,143 +64,106 @@ def get_table_data_for_year_month(
             sum_total_users[user] += amount
 
-        flw_amount_paid_data[d_name] = sum(sum_total_users.values())
         # take atleast 1 top user in cases where this variable is 0
         top_five_percent_len = len(sum_total_users) // 20 or 1
-        avg_top_flw_amount_paid_data[d_name] = sum(
-            sorted(sum_total_users.values(), reverse=True)[:top_five_percent_len]
-        )
-
-    if group_by_delivery_type:
-        for delivery_type_name in user_count_data.keys():
-            nm_amount_earned = nm_amount_earned_data.get(delivery_type_name, 0)
-            nm_amount_paid = nm_amount_paid_data.get(delivery_type_name, 0)
-            data.append(
-                {
-                    "delivery_type": delivery_type_name,
-                    "month": (month, year),
-                    "users": user_count_data[delivery_type_name],
-                    "services": service_count_data[delivery_type_name],
-                    "avg_time_to_payment": avg_time_to_payment_data.get(delivery_type_name),
-                    "max_time_to_payment": max_time_to_payment_data.get(delivery_type_name),
-                    "flw_amount_earned": flw_amount_earned_data.get(delivery_type_name, 0),
-                    "flw_amount_paid": flw_amount_paid_data.get(delivery_type_name, 0),
-                    "nm_amount_earned": nm_amount_earned,
-                    "nm_amount_paid": nm_amount_paid,
-                    "nm_other_amount_paid": nm_other_amount_paid_data.get(delivery_type_name, 0),
-                    "avg_top_paid_flws": avg_top_flw_amount_paid_data.get(delivery_type_name, 0),
-                }
-            )
-
-    else:
-        nm_amount_earned = sum(nm_amount_earned_data.values())
-        nm_amount_paid = sum(nm_amount_paid_data.values())
-        data.append(
-            {
-                "delivery_type": "All",
-                "month": (month, year),
-                "users": sum(user_count_data.values()),
-                "services": sum(service_count_data.values()),
-                "avg_time_to_payment": mean(avg_time_to_payment_data.values() or [0]),
-                "max_time_to_payment": max(max_time_to_payment_data.values() or [0]),
-                "flw_amount_earned": sum(flw_amount_earned_data.values()),
-                "flw_amount_paid": sum(flw_amount_paid_data.values()),
-                "nm_amount_earned": nm_amount_earned,
-                "nm_amount_paid": nm_amount_paid,
-                "nm_other_amount_paid": sum(nm_other_amount_paid_data.values()),
-                "avg_top_paid_flws": sum(avg_top_flw_amount_paid_data.values()),
-            }
-        )
-    return data
+        flw_amount_paid = sum(sum_total_users.values())
+        avg_top_paid_flws = sum(sorted(sum_total_users.values(), reverse=True)[:top_five_percent_len])
+        visit_data_dict[group_key].update({"flw_amount_paid": flw_amount_paid, "avg_top_paid_flws": avg_top_paid_flws})
+    return list(visit_data_dict.values())
```

Comment:
This method is getting quite longer, it becomes difficult to read/modify latter. It will add value to refactor it into a sort of class with separate methods for various colulms/queries.

For e.g. 
```
class VisitDataHelper:

def get_data(**filters):

def get_filter_kwargs()

def get_payment_data()

def get_visit_data()
```

Not exactly like that, but whatever to break it into smaller pieces will improve readability.

Reply:
I will try to address this in a future PR.

Reply:
plus one to this comment. not blocking but this is a very hard method to follow and review